python/cudf/cudf/core/dataframe.py

```diff
@@ -3091,16 +3094,17 @@ def where(self, cond, other=None, inplace=False):
                     col, other_col, cond_col
                 )
 
-                out[name] = _make_categorical_like(result, self._data[name])
+                out.append(_make_categorical_like(result, self._data[name]))
             else:
                 out_mask = cudf._lib.null_mask.create_null_mask(
                     len(col),
                     state=cudf._lib.null_mask.MaskState.ALL_NULL,
                 )
-                out[name] = col.set_mask(out_mask)
+                out.append(col.set_mask(out_mask))
 
         return self._mimic_inplace(
-            self._from_data_like_self(out), inplace=inplace
+            self._from_data_like_self(self._data._from_columns_like_self(out)),
```

Comment:
I still see a number of instances of `_from_data_like_self`. Can we use `_from_data` instead consistently now?

Reply:
> Can we use _from_data instead consistently now?

Getting there but not quite yet due to preserving the `index`. `IndexedFrame._from_data_like_self` is like `_from_data` but preserves the existing index which `IndexedFrame._from_data` doesn't do by default.